docker/pytorch/cpu-only/Dockerfile

```diff
@@ -0,0 +1,59 @@
+# syntax = docker/dockerfile:experimental
+#
+# NOTE: To build this you will need a docker version > 18.06 with
+#       experimental enabled and DOCKER_BUILDKIT=1
+#
+#       For reference:
+#           https://docs.docker.com/develop/develop-images/build_enhancements/
+#
+# This Dockerfile will build Docker Image with PyTorch+MKLDNN+BLIS and Torchvision installed for CPU only
+#
+# BLIS_URL can be provided by user while building the Docker Image (By Default, it is set to FLAME BLIS git repo)
+# Example commandline to build PyTorch with AMD BLIS
+# DOCKER_BUILDKIT=1 docker build --build-arg BLIS_URL=https://github.com/amd/blis.git . -t Docker-Image-Repo-Name
+ARG BASE_IMAGE=ubuntu:18.04
+ARG PYTHON_VERSION=3.7
+
+FROM ${BASE_IMAGE} as dev-base
+RUN --mount=type=cache,id=apt-dev,target=/var/cache/apt \
+    apt-get update && apt-get install -y --no-install-recommends \
+        build-essential \
+        ca-certificates \
+        ccache \
+        cmake \
+        curl \
+        git \
+        libjpeg-dev \
+        libpng-dev \
+        vim \
+        wget && \
+    rm -rf /var/lib/apt/lists/*
+RUN /usr/sbin/update-ccache-symlinks
+RUN mkdir /opt/ccache && ccache --set-config=cache_dir=/opt/ccache
+ENV PATH /opt/conda/bin:$PATH
+
+FROM dev-base as conda
+RUN curl -v -o ~/miniconda.sh -O  https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh  && \
+    chmod +x ~/miniconda.sh && \
+    ~/miniconda.sh -b -p /opt/conda && \
+    rm ~/miniconda.sh && \
+    /opt/conda/bin/conda install -y python=${PYTHON_VERSION} conda-build && \
+    /opt/conda/bin/conda install -y nomkl pyyaml numpy ipython ninja setuptools cmake cffi typing future && \
+    /opt/conda/bin/conda clean -ya
+
+ARG BLIS_URL=https://github.com/flame/blis.git
+# Download BLIS and place the necessary library and include files at /usr/local/lib and /usr/local/include respectively
+RUN cd ~ && git clone ${BLIS_URL} &&\
+    cd ~/blis && ./configure --prefix=~/BLISBuild --enable-cblas --enable-threading=openmp auto && make -j && make install && \
```

Comment:
There must be a better way to install this than specifying `--prefix=~/BLISBuild` and then manually copying some files from that directory. Does `--prefix=/usr/local/` or something like that not work?

Reply:
Thanks for the suggestions. I've made necessary changes to Dockerfile by setting BLIS_HOME to BLISBuild instead of copying files manually to /usr/local. Kindly review the updated file. 
I've been unable to fix the annotation failure issue with flake8-py3 (tests are failing). Requesting you to kindly help me to fix the issue.

Reply:
Don't worry about linter, it's flaky from upstream sometimes.

Concerning the actual changes - I guess it's OK to push it into `/root` but then I still don't understand the `cp` following it. AFAIK BLIS install sets its links up properly itself.

Reply:
Thanks for reviewing the changes. Regarding the cp line:
if [ ! -e /root/BLISBuild/lib/libblis.so ] ; then cp /root/BLISBuild/lib/libblis*.so /root/BLISBuild/lib/libblis.so ; fi
This line is added to make sure libblis.so is available under BLIS_HOME/lib. When FLAME BLIS is built from source, libblis.so is generated (This copy is not needed). However when AMD BLIS is built with MT, libblis-mt.so is generated (There is no libblis.so). If libblis.so is not found under BLIS_HOME/lib, it gives the below error as it is unable to find blis:
-------------------------------------
 CMake Warning at cmake/Dependencies.cmake:164 (message):
#14 268.8   Preferred BLAS (blis) cannot be found, now searching for a general BLAS
#14 268.8   library

 MKL library not found
#14 269.0 -- Checking for [blis]
#14 269.0 --   Library blis: BLAS_blis_LIBRARY-NOTFOUND
#14 269.0 -- Checking for [Accelerate]
#14 269.0 --   Library Accelerate: BLAS_Accelerate_LIBRARY-NOTFOUND
#14 269.0 -- Checking for [vecLib]
#14 269.0 --   Library vecLib: BLAS_vecLib_LIBRARY-NOTFOUND
#14 269.0 -- Checking for [openblas]
#14 269.0 --   Library openblas: /opt/conda/lib/libopenblas.so
#14 269.0 -- Looking for sgemm_
#14 269.1 -- Looking for sgemm_ - found

CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
#14 285.2 Please set them or make sure they are set and tested correctly in the CMake files:
#14 285.2 BLIS_LIB (ADVANCED)
#14 285.2     linked by target "torch_cpu" in directory /root/pytorch/caffe2
#14 285.2
#14 285.2 -- Configuring incomplete, errors occurred!
------------------------------

Reply:
OK, I see. Sounds like AMD's fork should be improved (I'd thinkthat a softlink to libblis.so should be made). Anyhow, for the time being it is what it is.